compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -553,13 +639,15 @@ static constexpr unsigned cudaWarpSize = 32;
 static bool matchGPUReduction(linalg::LinalgOp op, TileSizesListType &tileSizes,
                               std::array<int64_t, 3> &workgroupSize) {
   // TODO: match the sequence the strategy supports.
+  auto fill = m_StructuredOp<linalg::FillOp>();
```

Comment:
Side note, as we iterate, I would prob. separate the dims into a separate struct to try and get to an API that resembles.

```
auto cstrs = OpConstraints().dim(...)
                            .dim(...)
                            .dim(...)
                            .input(AllOperands(), IsPermutation())
                            ...;
m_StructuredOp(Cstrs).witOperands(Any(), Any(), fill);
```

Reply:
or: `m_StructuredOp(Cstrs).witOperands(/*inputs=*/{Any(), Any()}, /*outputs=*/{fill});`